### ISSUE TITLE
girder_client.IncompleteResponseError

### ISSUE THREAD

**Issue by doctorcolossus** [girder/girder#3371]
I'm experiencing a weird problem where the downloaded file is larger than it should be, which raises the `girder_client.IncompleteResponseError` exception. I have tried uploading the file both via `girder_client` and via the web interface. The web interface shows the correct size in bytes. However, downloading either through the web interface or with `girder_client` results in a file which is larger than the original and has an incorrect `md5sum`. I am not sure what else to try.

```
$ girder-client \
>   --api-url {api_url} \
>   --api-key {api_key} \
>   download {identifier} \
>   libmyplugins.so
[####################################]  234.18k/0.00  100%  libmyplugins.so

Traceback (most recent call last):
  File "/usr/local/bin/girder-client", line 8, in <module>
    sys.exit(main())
  File "/usr/lib/python3/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/girder_client/cli.py", line 303, in _download
    gc.downloadResource(parent_id, local_folder, parent_type)
  File "/usr/local/lib/python3.6/dist-packages/girder_client/__init__.py", line 1348, in downloadResource
    self.downloadFolderRecursive(resourceId, dest, sync)
  File "/usr/local/lib/python3.6/dist-packages/girder_client/__init__.py", line 1325, in downloadFolderRecursive
    self.downloadItem(item['_id'], dest, name=item['name'])
  File "/usr/local/lib/python3.6/dist-packages/girder_client/__init__.py", line 1262, in downloadItem
    created=files[0]['created'])
  File "/usr/local/lib/python3.6/dist-packages/girder_client/__init__.py", line 1200, in downloadFile
    raise IncompleteResponseError('File %s download' % fileId, fileObj['size'], size)
girder_client.IncompleteResponseError: File 61af18532a2cd16e8410b3b1 download (239799 of 239792 bytes received)
```

The `girder_client` version is 3.1.7. I don't know about the girder server version. I am not the girder server administrator. If there is a way that I can find that out as a user, please let me know. Thank you.

**Comment by zachmullen** [comment]
That's quite strange. I'd like to reproduce this issue, but I suspect I'd need to use the same file you have, since I've never seen this with other files. Is it one you'd be able to share?

**Comment by doctorcolossus** [comment]
Sure, just let me know how to send it to you.

**Comment by doctorcolossus** [comment]
Okay, I managed to resolve this. I figured out that the girder API URL I had been using was going through an nginx reverse proxy. After I was able to find the direct URL which bypassed that proxy, dowloading worked correctly. It is strange that using Girder through this proxy only seemed to cause issues with this one file and not others... 🤔 It was `libmyplugins.so`, generated by [TensorRTX](https://github.com/wang-xinyu/tensorrtx). I have posted it [here](https://www.file.io/download/KX9IC6e4B2uM) in case you wish to investigate.
